common/lib/transport/connectionmanager.js

```diff
@@ -422,6 +422,14 @@ var ConnectionManager = (function() {
 		if(existingState == this.states.closing || existingState == this.states.closed)
 			return false;
 
+		/* if the transport is not connected (eg because it failed during a
+		 * scheduleTransportActivation#onceNoPending wait) then don't activate it */
+		if(!transport.isConnected) {
+			Logger.logAction(Logger.LOG_MINOR, 'ConnectionManager.activateTransport()', 'Declining to activate transport ' + transport + ' since it appears to no longer be connected');
+			Utils.arrDeleteValue(this.pendingTransports, transport);
```

Comment:
I think I would have reversed the order of these things; ie unconditionally remove from the `pendingTransports` list, then  decline to activate + return if not connected.